opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/Configuration.java

```diff
@@ -847,25 +848,31 @@ public void setProjects(Map<String, Project> projects) {
     }
 
     /**
-     * Adds a group to the set. This is performed upon configuration parsing
+     * Adds a group to the map. This is performed upon configuration parsing
      *
      * @param group group
      * @throws IOException when group is not unique across the set
      */
     public void addGroup(Group group) throws IOException {
-        if (!groups.add(group)) {
+        if(!groups.containsKey(group.getName())){
```

Comment:
Wrong formatting, checkstyle will probably fail. Please try to compile the code locally by `./mvnw compile`.

Reply:
also the condition is wrong